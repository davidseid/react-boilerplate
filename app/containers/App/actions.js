import {
  SAVE_ENTRY,
  SAVE_ENTRY_SUCCESS,
  SAVE_ENTRY_ERROR,
} from './constants';

export function saveEntry() {
  return {
    type: SAVE_ENTRY,
  };
}

export function entrySaved() {
  return {
    type: SAVE_ENTRY_SUCCESS,
  };
}

export function entrySavingError(error) {
  return {
    type: SAVE_ENTRY_ERROR,
    error,
  };
}
