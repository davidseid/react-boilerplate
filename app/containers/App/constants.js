/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const DEFAULT_LOCALE = 'en';
export const SAVE_ENTRY = 'boilerplate/App/SAVE_ENTRY';
export const SAVE_ENTRY_SUCCESS = 'boilerplate/App/SAVE_ENTRY_SUCCESS';
export const SAVE_ENTRY_ERROR = 'boilerplate/App/SAVE_ENTRY_ERROR';
export const LOAD_JOURNAL = 'boilerplate/App/LOAD_JOURNAL';
export const LOAD_JOURNAL_SUCCESS = 'boilerplate/App/LOAD_JOURNAL_SUCCESS';
export const LOAD_JOURNAL_ERROR = 'boilerplate/App/LOAD_JOURNAL_ERROR';
