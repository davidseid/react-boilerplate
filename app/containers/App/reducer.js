import { fromJS } from 'immutable';

import {
  SAVE_ENTRY_SUCCESS,
  SAVE_ENTRY,
  SAVE_ENTRY_ERROR,
} from './constants';

const initialState = fromJS({
  loading: false,
  saving: false,
  error: false,
  journal: false,
});

function appReducer(state = initialState, action) {
  switch (action.type) {
    case SAVE_ENTRY:
      return state
        .set('saving', true)
        .set('error', false);
    case SAVE_ENTRY_SUCCESS:
      return state
        .set('saving', false);
    case SAVE_ENTRY_ERROR:
      return state
        .set('error', action.error)
        .set('saving', false);
    default:
      return state;
  }
}

export default appReducer;
