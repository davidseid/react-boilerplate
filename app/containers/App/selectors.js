import { createSelector } from 'reselect';

const selectGlobal = (state) => state.get('global');

const selectRoute = (state) => state.get('route');

const makeSelectLoading = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('loading')
);

const makeSelectSaving = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('saving')
);

const makeSelectError = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('error')
);

const makeSelectJournal = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('journal')
);

const makeSelectLocation = () => createSelector(
  selectRoute,
  (routeState) => routeState.get('location').toJS()
);

export {
  selectGlobal,
  makeSelectLoading,
  makeSelectSaving,
  makeSelectError,
  makeSelectJournal,
  makeSelectLocation,
};
