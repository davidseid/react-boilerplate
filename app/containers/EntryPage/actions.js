import {
  CHANGE_ENTRY,
} from './constants';

export function changeEntry(entry) {
  return {
    type: CHANGE_ENTRY,
    entry,
  };
}
