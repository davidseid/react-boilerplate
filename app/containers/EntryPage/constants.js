export const CHANGE_ENTRY = 'boilerplate/EntryPage/CHANGE_ENTRY';
