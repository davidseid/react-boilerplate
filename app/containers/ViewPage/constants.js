/*
 *
 * ViewPage constants
 *
 */

export const DEFAULT_ACTION = 'app/ViewPage/DEFAULT_ACTION';
