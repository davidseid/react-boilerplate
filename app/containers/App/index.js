/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';

import EntryPage from 'containers/EntryPage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import NavBar from 'components/NavBar';
import CenteredDiv from 'components/CenteredDiv';

export default function App() {
  return (
    <CenteredDiv>
      <NavBar />
      <Switch>
        <Route exact path="/" component={EntryPage} />
        <Route component={NotFoundPage} />
      </Switch>
    </CenteredDiv>
  );
}
