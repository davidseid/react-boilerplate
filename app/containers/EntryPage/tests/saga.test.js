import { put, takeLatest } from 'redux-saga/effects';

import { SAVE_ENTRY } from 'containers/App/constants';
import { entrySaved, entrySavingError } from 'containers/App/actions';

import saveData, { storeEntry } from '../saga';

const entry = 'Sample Entry';

/* eslint-disable redux-saga/yield-effects */
describe('storeEntry Saga', () => {
  let storeEntryGenerator;

  beforeEach(() => {
    storeEntryGenerator = storeEntry();

    const selectDescriptor = storeEntryGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = storeEntryGenerator.next(entry).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the entrySaved action if it saves the data successfully', () => {
    const putDescriptor = storeEntryGenerator.next().value;
    expect(putDescriptor).toEqual(put(entrySaved()));
  });

  it('should call the entrySavingError action if the action errors', () => {
    const response = new Error('Some error');
    const putDescriptor = storeEntryGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(entrySavingError(response)));
  });
});

describe('saveData Saga', () => {
  const saveDataSaga = saveData();

  it('should start task to watch for SAVE_ENTRY action', () => {
    const takeLatestDescriptor = saveDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(SAVE_ENTRY, storeEntry));
  });
});
