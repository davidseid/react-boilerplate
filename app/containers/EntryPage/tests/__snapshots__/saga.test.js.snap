// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storeEntry Saga should call the entrySavingError action if the action errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`storeEntry Saga should call the entrySavingError action if the action errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3000/api/journal",
      Object {
        "entry": "Sample Entry",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`storeEntry Saga should dispatch the entrySaved action if it saves the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`storeEntry Saga should dispatch the entrySaved action if it saves the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3000/api/journal",
      Object {
        "entry": "Sample Entry",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
